zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -156,59 +237,74 @@ impl PolicyEnforcer {
                     }
                 } else {
                     // check for undefined values in rules and initialize them to defaults
-                    for (rule_offset, rule) in rules.iter_mut().enumerate() {
-                        if rule.interfaces.is_none() {
-                            tracing::warn!("ACL config interfaces list is empty. Applying rule #{} to all network interfaces", rule_offset);
-                            rule.interfaces =
-                                Some(get_interface_names_by_addr(Ipv4Addr::UNSPECIFIED.into())?);
+                    for rule in rules.iter_mut() {
+                        if rule.id.trim().is_empty() {
+                            bail!("Found empty rule id in rules list");
                         }
                         if rule.flows.is_none() {
-                            tracing::warn!("ACL config flows list is empty. Applying rule #{} to both Ingress and Egress flows", rule_offset);
+                            tracing::warn!("Rule '{}' flows list is not set. Setting it to both Ingress and Egress", rule.id);
                             rule.flows =
                                 Some([InterceptorFlow::Ingress, InterceptorFlow::Egress].into());
                         }
-                        if rule.usernames.is_none() {
-                            rule.usernames = Some(Vec::new());
+                    }
+                    // check for undefined values in subjects and initialize them to defaults
+                    for subject in subjects.iter_mut() {
+                        if subject.id.trim().is_empty() {
+                            bail!("Found empty subject id in subjects list");
+                        }
+                        if subject.interfaces.is_none() {
+                            if subject.cert_common_names.is_none() && subject.usernames.is_none() {
+                                tracing::warn!(
+                                    "Subject '{}' is empty. Setting it to wildcard (all network interfaces)",
+                                    subject.id
+                                );
+                                subject.interfaces = Some(get_interface_names_by_addr(
```

Comment:
If `IPv4` is disabled in the system this operation will fail. Check first `IPv6` and then `IPv4`, return error when both fail.

Reply:
Following @fuzzypixelz's implementation of wildcards, this interface scanning is no longer required.